src/main/java/com/visenze/productsearch/ProductSearch.java

```diff
@@ -0,0 +1,208 @@
+package com.visenze.productsearch;
+
+import com.visenze.productsearch.param.*;
+import com.visenze.visearch.ClientConfig;
+import com.visenze.visearch.ResponseMessages;
+import com.visenze.visearch.internal.InternalViSearchException;
+import com.visenze.visearch.internal.http.ViSearchHttpClientImpl;
+import com.visenze.visearch.internal.http.ViSearchHttpResponse;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+/**
+ * <h1> ProductSearch </h1>
+ * This class wraps ViSearch internally to help delegate tasks and re-use
+ * modular pieces that were included in the com.visenze.visearch package. Since
+ * This class is located in separate package, there will be a lot of dependency
+ * from the .visearch package.
+ * <p>
+ * Future plan? To split things that are
+ * non-api specific into a 'common' or 'util' package for both ProductSearch and
+ * ViSearch to use.
+ *
+ * @author Shannon Tan
+ * @version 1.0
+ * @since 08 Jan 2021
+ */
+public class ProductSearch {
+
+    /**
+     * Default endpoint if none is set
+     */
+    public static final String DEFAULT_ENDPOINT = "https://search-dev.visenze.com//v1";
```

Comment:
default endpoint is `https://search.visenze.com/v1` . the `-dev` is staging endpoint, not production